backend/src/test/java/ca/mcgill/purposeful/features/ID015_createIdeaStepDefinitions.java

```diff
@@ -0,0 +1,233 @@
+          String inprogress,
+          String isprivate,
+          String domains,
+          String topics,
+          String techs,
+          String iconurl,
+          String supportingimageurls)
+          throws Throwable {
+    Boolean isPaid = Boolean.parseBoolean(ispaid);
+    Boolean inProgress = Boolean.parseBoolean(inprogress);
+    Boolean isPrivate = Boolean.parseBoolean(isprivate);
+    ArrayList<String> domainIds = new ArrayList<String>();
+    ArrayList<String> techIds = new ArrayList<String>();
+    ArrayList<String> topicIds = new ArrayList<String>();
+    ArrayList<String> supportingImgIds = new ArrayList<String>();
+    String iconUrlId = "";
+
+    String[] domainStr = domains.split(",");
+    String[] techStr = techs.split(",");
+    String[] topicStr = topics.split(",");
+    String[] urlStr = supportingimageurls.split(",");
+
+    for (String domain_name : domainStr) {
+      String domainId = "";
+      for (Domain domain : domainRepository.findAll()) {
+        if (domain.getName().equals(domain_name)) {
+          domainId = domain.getId();
+          break;
+        }
+      }
+      domainIds.add(domainId);
+    }
+    for (String tech_name : techStr) {
+      for (Technology tech : technologyRepository.findAll()) {
+        if (tech.getName().equals(tech_name)) {
+          techIds.add(tech.getId());
+          break;
+        }
+      }
+    }
+    for (String topic_name : topicStr) {
+      String topicId = "";
+      for (Topic topic : topicRepository.findAll()) {
+        if (topic.getName().equals(topic_name)) {
+          topicId = topic.getId();
+          break;
+        }
+      }
+      topicIds.add(topicId);
+    }
+    for (String url_str : urlStr) {
+      for (URL url : urlRepository.findAll()) {
+        if (url.getURL().equals(url_str)) {
+          supportingImgIds.add(url.getId());
+          break;
+        }
+      }
+    }
+    for (URL url : urlRepository.findAll()) {
+      if (url.getURL().equals(iconurl)) {
+        iconUrlId = url.getId();
+        break;
+      }
+    }
+    IdeaRequestDTO ideaDto =
+        new IdeaRequestDTO(
+            null,
+            title,
+            purpose,
+            description,
+            null,
+            isPaid,
+            inProgress,
+            isPrivate,
+            domainIds,
+            techIds,
+            topicIds,
+            supportingImgIds,
+            iconUrlId);
+
+    // make a post request to create the idea and store the response
+    HttpEntity<IdeaRequestDTO> requestEntity =
+        new HttpEntity<>(ideaDto, cucumberUtil.bearerAuthHeader(this.jwtToken));
+    try {
+      this.response =
+          client.exchange("/api/idea/create", HttpMethod.POST, requestEntity, IdeaRequestDTO.class);
+    } catch (Exception e) {
+      this.response =
+          client.exchange("/api/idea/create", HttpMethod.POST, requestEntity, String.class);
+    }
+  }
+
+  @Then("^a new idea exists in the database with title (.*), icon (.*)$")
```

Comment:
It would be great if you could verify more fields than just title and icon. The best would be to verify all the fields for an idea

Reply:
Initially i verified all fields, but shidan told me it was unnecessary. Shall i add them anyway?